src/base/Optional.h

```diff
@@ -60,10 +73,33 @@ class Optional
         return *this;
     }
 
+    void clear() {
+        if (hasValue_) {
+            value_.~Value();
+            hasValue_ = false;
```

Comment:
To avoid (unlikely) infinite loops and other problems if Value destructor has side effects:

```suggestion
            hasValue_ = false;
            value_.~Value();
```

Reply:
Done (280c5f8).